### PR TITLE
[CI:BUILD] Contrib: Add containerfile to create podman-remote binary image

### PR DESCRIPTION
**Pull request by praveenkumar**
Try to partial address #14664

```release-note
- Add container file to build remote binary for each platform
- Add README.md file around same
```

Longer term plan is to have it attach either to quay and add github trigger to generate the image for each 4.x tag/branch. Only issue is as of now quay github trigger can only generate `amd64` images not the `arm64` one so may be we need to attach it to current cirrus CI to generate image for different arch and push to quay, same way we are doing for `quay.io/containers/podman`.

Another gap is created binaries for mac/windows are not signed one and we need to check if user try to copy those binary from the container image does it showing the signing warning.

### CODE REVIEW

File: contrib/podmanremoteimage/README.md
Comment (flouthoc) [inline contrib/podmanremoteimage/README.md:20]:
```suggestion
$ podman cp $(podman create --name remote-temp quay.io/containers/podman-remote-artifacts:latest):/podman-remote-static . && podman rm remote-temp
```

Comment (flouthoc) [reply contrib/podmanremoteimage/README.md:20]: Points to personal repo, are we expecting to create similar tag in `containers`

Comment (flouthoc) [reply contrib/podmanremoteimage/README.md:20]: I think similar comment goes for `darwin`

**Review by flouthoc — APPROVED** [review]
LGTM other than two nits above.

File: contrib/podmanremoteimage/README.md
Comment (praveenkumar) [reply contrib/podmanremoteimage/README.md:20]: @flouthoc yes I forgot to remove my personal repo :) . I think we should have similar repo/tag in `containers` org for consumption.

**Comment by rhatdan** [comment]
@mheon @cevich @vrothberg PTAL

File: contrib/podmanremoteimage/README.md
Comment (vrothberg) [inline contrib/podmanremoteimage/README.md:9]:
```suggestion
for each PR merged (for any branch or tag).
```

**Review by vrothberg — COMMENTED** [review]
What is the benefit over curl-ing the binaries from the release pages?  I assume for testing purposes but I'd appreciate making the intention explicit in the README.

I'd like some more maintainers to have a look.

**Comment by cevich** [comment]
> Longer term plan is to have it attach either to quay and add github trigger to generate the image for each 4.x tag/branch.

Yeah, can't use quay.  We have two Cirrus-CI tasks setup to build multi-arch image manifests, `image_build` and `test_image_build`.  The later is intended for PRs (like this one) so you can test out new/changed builds.  It's only available for manual-trigger, and only if you put `[CI:BUILD]` in the PR description (which I just did).

**Comment by cevich** [comment]
So on your next force-push, you'll see the `test_image_build` manual-trigger jobs pop up.  I'll most likely need to help add the new one, since [the build script is sensitive to the directory names.](https://github.com/containers/automation_images/blob/main/build-push/bin/main.sh#L57-L68).  I'll also need to setup the new repo in quay for it to push to.

File: contrib/podmanremoteimage/Containerfile
Comment (cevich) [inline contrib/podmanremoteimage/Containerfile:1]: All of our other container images are based on `registry.fedoraproject.org/fedora:latest`, for consistency it would be good to use that here too.  An alternative could be `registry.fedoraproject.org/fedora-minimal:latest`.  In either case, check out the `contrib/podmanimage/*/Containerfiles` to see how we install `podman`.  It's _MUCH_ faster to install than to compile during image-build time (due to needing emulation).

File: contrib/podmanremoteimage/README.md
Comment (cevich) [inline contrib/podmanremoteimage/README.md:9]: The current build script probably won't work for mac/windows since it runs on a Fedora VM w/ nested-virt emulation (for multi-arch).  Though I could be wrong, I'm assuming we need a mac or windows VM to build for those platforms, no?

File: contrib/podmanremoteimage/README.md
Comment (cevich) [inline contrib/podmanremoteimage/README.md:26]: Oh! nevermind my last comment, the above can easily be accommodated in our build script.  It has [an image-mod hook available](https://github.com/containers/automation/tree/main/build-push#use-in-automation-with-modified-images) (docs) that can produce these extra image flavors prior to pushing.

For reference, [we currently use a version-tagging mod-script](https://github.com/containers/automation_images/blob/main/build-push/bin/tag_version.sh) to handle creation of extra tags for the `stable` podman images.

Comment (cevich) [reply contrib/podmanremoteimage/README.md:26]: (I can do this work later, no need to make this PR depend on it.

File: contrib/podmanremoteimage/README.md
Comment (cevich) [reply contrib/podmanremoteimage/README.md:9]: Perhaps it's best not to mention "plans" in readme files.  Better to just file an RFE issue so the intent isn't lost.

File: contrib/podmanremoteimage/Containerfile
Comment (cevich) [reply contrib/podmanremoteimage/Containerfile:1]: errr...nevermind :disappointed: that won't work for Mac or Windows will it?  My concern is the multi-arch compiles are extremely slow under emulation.  Hmmmm.  Perhaps the thing to do is just grab the binaries directly from the CI system's `artifacts` task.  They're not signed, but will be posted continuously for `main`, and recent release-branches (where we added ARM64 builds).

Comment (praveenkumar) [reply contrib/podmanremoteimage/Containerfile:1]: @cevich Does it also published for specific tags like rc one? One of the use case which I am trying to solve with it, if we do any release on github (with specific branch) be it alpha, beta, rc we should have release binaries for those bits (longer term) either through the container image or from the CI artifacts. If I see https://github.com/containers/podman/pull/15202 (which is the PR for rc3) then looks like we just use the rc3 commit to do the tagging and there is no CI artifact for it.

File: contrib/podmanremoteimage/README.md
Comment (praveenkumar) [reply contrib/podmanremoteimage/README.md:9]: @cevich Even for podman-remote binaries we need windows/mac vm? I think it is cross compiled so should work on Fedora VM w/ nested-virt emulation (might be slow but should work)?

File: contrib/podmanremoteimage/Containerfile
Comment (cevich) [reply contrib/podmanremoteimage/Containerfile:1]: Mostly, I don't think tags to, but for sure on most recent branches (and going forward) the CI system continuously publishes artifacts at a consistent URL.  The URL will be of the form specified here: https://cirrus-ci.org/guide/writing-tasks/#latest-build-artifacts

File: contrib/podmanremoteimage/README.md
Comment (cevich) [reply contrib/podmanremoteimage/README.md:9]: Nope, my understanding is wrong, [see my next comment below](https://github.com/containers/podman/pull/15813#pullrequestreview-1109266061).

File: contrib/podmanremoteimage/Containerfile
Comment (cevich) [inline contrib/podmanremoteimage/Containerfile:6]: What do you think of: Instead of compiling, curl the latest binaries from CI system using the branch-name specified as a build-arg?

Comment (praveenkumar) [reply contrib/podmanremoteimage/Containerfile:6]: @cevich if we are adding this to CI for each PR then first we have to wait for other CI run to be succeed (to generate required images) as part of some step builds?

Comment (cevich) [reply contrib/podmanremoteimage/Containerfile:6]: Not the way the builds run today.  They're optional on `[CI:BUILD]` PRs (to support testing/development).  Otherwise they only run via a daily cron-job on `main`.

File: contrib/podmanremoteimage/README.md
Comment (TomSweeneyRedHat) [inline contrib/podmanremoteimage/README.md:7]: Nit and way past, but just in case containerfile -> Containerfile

File: contrib/podmanremoteimage/README.md
Comment (TomSweeneyRedHat) [inline contrib/podmanremoteimage/README.md:10]: following -> the following

**Pull request merged**
